### PR TITLE
flexible user agent getHeader filtering by env var

### PR DESCRIPTION
**Pull request by metachris**
## 📝 Summary

- Allows setting filtered user agents through env var `NO_HEADER_USERAGENTS`
- Test for getHeader user-agent filter

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`
* [x] I have seen and agree to `CONTRIBUTING.md`

### CODE REVIEW

File: services/api/service.go [hunk 1]
```diff
@@ -1398,7 +1403,7 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 	}
 
 	// Prepare the response data
-	getHeaderResponse, err := BuildGetHeaderResponse(payload, api.blsSk, api.publicKey, api.opts.EthNetDetails.DomainBuilder)
+	getHeaderResponse, err := BuildDummyGetHeaderResponse(payload, api.blsSk, api.publicKey, api.opts.EthNetDetails.DomainBuilder)
```
Comment (michaelneuder) [inline services/api/service.go:1406]: what is the value of building a dummy here ?

Comment (michaelneuder) [reply services/api/service.go:1406]: maybe just the name Dummy makes it sound like a test. maybe Empty is better ?

**Review by michaelneuder — DISMISSED** [review]

File: services/api/service.go (hunk 1 above)
Comment (metachris) [reply services/api/service.go:1406]: Just wanted to make it abundantly clear this is not producing a valid payload, just the necessary bits that the Redis stuff needs. Empty works too, makes sense

Comment (metachris) [reply services/api/service.go:1406]: done

**Pull request merged**